pkg/local-resource-manager/resource_services.go

```diff
@@ -0,0 +1,153 @@
+		for _, metrics := range nodesMetrics.Items {
+			if node.Name != metrics.Name {
+				// So that we can select just the nodes that we want
+				continue
+			}
+
+			metricsStruct := getNodeResourceMetrics(&metrics, &node)
+			nodeInfo := getNodeInfo(&node, metricsStruct)
+			nodesInfo = append(nodesInfo, *nodeInfo)
+			uids = append(uids, string(node.UID))
+		}
+	}
+
+	return &nodesInfo, nil
+}
+
+// getNodeResourceMetrics gets a ResourceMetrics struct
+func getNodeResourceMetrics(nodeMetrics *metricsv1beta1.NodeMetrics, node *corev1.Node) *ResourceMetrics {
+	cpuTotal := node.Status.Allocatable.Cpu()
+	cpuUsed := nodeMetrics.Usage.Cpu()
+	memoryTotal := node.Status.Allocatable.Memory()
+	memoryUsed := nodeMetrics.Usage.Memory()
+	ephemeralStorage := nodeMetrics.Usage.StorageEphemeral()
+	return fromResourceMetrics(*cpuTotal, *cpuUsed, *memoryTotal, *memoryUsed, *ephemeralStorage)
+}
+
+// getNodeInfo gets a NodeInfo struct
+func getNodeInfo(node *corev1.Node, metrics *ResourceMetrics) *NodeInfo {
+	return fromNodeInfo(string(node.UID), node.Name, node.Status.NodeInfo.Architecture, node.Status.NodeInfo.OperatingSystem, *metrics)
+}
+
+// createFlavourCustomResource creates a new flavour custom resource
+func forgeFlavour(maxCPUAvailableNode NodeInfo, cl client.Client) error {
+	flavour := (&nodecorev1alpha1.Flavour{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      maxCPUAvailableNode.UID + "-" + discoverymanager.K8S_TYPE + "-" + generateUniqueString(maxCPUAvailableNode.UID), // Using pattern nodeID-Type-rand(4)
+			Namespace: "default",                                                                                                       // Specify the namespace where you want to create the CR
+		},
+		Spec: nodecorev1alpha1.FlavourSpec{
+			ProviderID: maxCPUAvailableNode.UID,
+			Type:       nodecorev1alpha1.K8S,
+			Characteristics: nodecorev1alpha1.Characteristics{
+				Cpu:              maxCPUAvailableNode.ResourceMetrics.CPUAvailable,
+				Memory:           maxCPUAvailableNode.ResourceMetrics.MemoryAvailable,
+				EphemeralStorage: maxCPUAvailableNode.ResourceMetrics.EphemeralStorage,
+			},
+			Policy: nodecorev1alpha1.Policy{
+				Partitionable: &nodecorev1alpha1.Partitionable{
+					CpuMin:     int(CPU_MIN),
+					MemoryMin:  int(MEMORY_MIN),
+					CpuStep:    int(CPU_STEP),
+					MemoryStep: int(MEMORY_STEP),
+				},
+				Aggregatable: &nodecorev1alpha1.Aggregatable{
+					MinCount: int(MIN_COUNT),
+					MaxCount: int(MAX_COUNT),
+				},
+			},
+			Owner: nodecorev1alpha1.NodeIdentity{
+				Domain: DOMAIN,
+				IP:     IP_ADDR,
+				NodeID: maxCPUAvailableNode.UID,
+			},
+			Price: nodecorev1alpha1.Price{
+				Amount:   AMOUNT,
+				Currency: CURRENCY,
+				Period:   PERIOD,
+			},
+			OptionalFields: nodecorev1alpha1.OptionalFields{
+				Availability: true,
+			},
+		},
+	})
+
+	err := cl.Create(context.Background(), flavour)
+	if err != nil {
+		return err
+	} else {
+		log.Printf("Flavour created successfully")
+		return nil
+	}
```

Comment:
this is a forge not a "forge and save on DB" function. Split this logic

Reply:
Logic split, fixed